include/net/net_if.h

```diff
@@ -231,12 +228,19 @@ struct net_if_ipv6 {
 
 	/** Retransmit timer (RFC 4861, page 52) */
 	u32_t retrans_timer;
+#if defined(CONFIG_NET_IPV6_ND)
+	/** Router solicitation timer node */
+	sys_snode_t rs_node;
 
-	/** IPv6 hop limit */
-	u8_t hop_limit;
```

Comment:
Why do we move the hop limit in this commit "net/iface: Coalesce all RS timers through one" outside of IPV6_ND? Does this belong to some other commit?

Reply:
It never was part of IPV6_ND, only RS. I did move it in order to get the RS-related attributes inside a new ifdef for IPV6_ND. It's fine here.